### PR TITLE
Add type safety to jest config

### PR DESCRIPTION
**Pull request by uladzimirdev**
### Description

Add type safety to jest config

image

### How to verify

jest tests should pass

### Checklist

~- [ ] Tests have been added/updated to cover changes in this PR~

### CODE REVIEW

File: jest.config.js
Comment (iethree) [inline jest.config.js:1]: can this work as `jest.config.ts` ? weird to do type checks on .js

the docs seem to imply you can: https://jestjs.io/docs/configuration

Comment (npretto) [reply jest.config.js:1]: It seems it requires us to install `ts-node` as devDep though:
> To read TypeScript configuration files Jest requires [ts-node](https://npmjs.com/package/ts-node). Make sure it is installed in your project.

Comment (uladzimirdev) [reply jest.config.js:1]: With .ts extension you need to transpile config every time, so using ts-check on js files is kinda a workaround without downsides of using ts extension

**Review by npretto — APPROVED** [review]
LGTM, should we backport it so that we won't have conflicts if we change the config in PRs related to bugfixes?

**Review by iethree — APPROVED** [review]

**Pull request merged**